### DISCUSSION TITLE
Error "Invalid GitHub Pages source folder."

### DISCUSSION THREAD

**Discussion by elena-alcaraz** [community/community#130206] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Estoy intentando publicar mi repositorio en GitHub Pages, pero siempre me da el error Invalid GitHub Pages source folder.

Tengo publicada la carpeta docs en la raíz del repositorio, y todo configurado correctamente. ¿Alguien sabe qué puede estar pasando? 
Gracias!

**Comment by gitstua** [comment]
Answering in English - Not sure if this is a public repo but can you check the folder specified in the workflow file? it should be similar to this... (note it is case sensitive so `docs` isn't the same as `Docs`) https://github.com/gitstua-labs/pages-demo/blob/7ec39d657d852216bedcb069ff98a6c85b5d13bc/.github/workflows/jekyll-gh-pages.yml#L36

You may also find help in the docs: https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site#creating-a-repository-for-your-site

**Comment by legndery** [comment]
Happening to me as well. I have created GH pages before many times never something like this happened. I tried using `docs` folder, tried using different gh-pages branch etc etc.

Mine is public forked repo: https://github.com/legndery/kanakanjiquiz

**Reply by elena-alcaraz** [reply]
Did you find the error? I cant explain what is happening, I've published several repos without problems, I've tried deleting the folder and creating it again, trying with different methods... Dont know whats happening

**Reply by legndery** [reply]
I literally have no idea why this is happening. No description or anything about the error. No google search result except for this exact forum post.

**Reply by legndery** [reply]
But, when I created the `docs` folder and tried to use it as source folder, somehow the `main` `/root` folder got deployed I have no idea how. 
So for now, my thing is deployed at `x.github.io/repo/docs` which is very lame but for now i will use it before more information comes out.

**Reply by beethogedeon** [reply]
I'm currently facing that issue too

**Comment by TyronNIKO** [comment]
Hi. Same here... here is deploy settings https://github.com/TyronNIKO/goit-js-hw-12/blob/main/.github/workflows/deploy.yml
Trying to select branch by default '/(root)' - Save btn is disabled - if select '/(root)' or else manualy - SAVE is active but when pressed - you get error **Invalid GitHub Pages source folder**

**Reply by gitstua** [reply]
👋🏽  Thanks for the detailed report. I have passed this information on to the team.

**Reply by ManuelBerrueta** [reply]
I'm getting the same behavior; it sounds like there is an issue here 😕

**Reply by ManuelBerrueta** [reply]
@gitstua do you guys need to create an incident for this? https://www.githubstatus.com/ still shows all 🟢 green 🤷‍♂️

**Reply by bigcat88** [reply]
> 👋🏽 Thanks for the detailed report. I have passed this information on to the team.

any news on this? without it currently we have no option to disable GitHub Pages feature for the repository :(

**Reply by ghost** [reply]
I am having the exact same issue!

**Comment by pukmajster** [comment]
Same issue here. The only time it went through is when i chose a branch without an index.html and it published the readme instead.

**Comment by JellyJoe13** [comment]
Hi there! Same for me. I have a separate branch for documentation and the content is in /docs with an index.html and .nojekyll file and the same error.

**Comment by nicu** [comment]
I'm getting the same error when trying to deploy a Docusaurus generated page. The `main` branch is selected by default and `/(root)`. When I select the `gh-pages` branch the Save button is disabled, I have to select `/docs` then `/(root)` again, press Save and I get the error: `Invalid GitHub Pages source folder.`

**Reply by nnthruslpn** [reply]
same error

**Comment by philippemiron** [comment]
I'm getting the same error. I tried deleting the branch and starting from a single `index.html` file and got the same error. Note that this is not happening on my "main" page (philippemiron.github.io), but another repo that should show up as a `philippemiron.github.io/xyz`.

**Comment by ManuelBerrueta** [comment]
Adding a +1. I am getting the same error as well. It won't let you change the source branch (the "Save" button is disabled) and thus won't deploy, as described by @TyronNIKO

**Comment by iCodeWebApps** [comment]
I am getting the same error!

I unpublished my site.  Then I was trying to getting it republished, it defaulted to main branch for Build and Deployment.  

When tried switching it to my gh-pages branch, it returned error: Invalid GitHub Pages source folder.

https://github.com/iCodeWebApps/dev1/settings/pages/source
Request Method:
POST
Status Code:
302 Found
Remote Address:
20.205.243.166:443
Referrer Policy:
strict-origin-when-cross-origin

**Comment by foochris** [comment]
Same issue as the above posts.

@gitstua Please escalate this issue to the GitHub pages team. The [GitHub Status page](https://www.githubstatus.com) indicates that Pages is `normal`, however, with the above comments, there certainly appears to be an issue.

**Comment by Samska** [comment]
Same issue here, I unpublished a project that I already have the GitHub Pages working fine and tried to publish now I got the same error.

**Comment by bkleinen** [comment]
Same here. Exact thing that worked yesterday doesn't work today:

fork this repo, enable actions, run gh-pages-fork.yml, publish from gh-pages.
https://github.com/htw-imi-showtime/showtime-website

workflow: https://github.com/htw-imi-showtime/showtime-website/blob/main/.github/workflows/gh-pages-fork.yml

**Comment by Mikopet** [comment]
same here. cannot change root to `docs` branch.

**Reply by drjzhn** [reply]
> same here. cannot change root to `docs` branch.

Same issue

**Comment by MariaYminova** [comment]
The same problem

**Comment by sksksk2024** [comment]
this needs to be resolved asap 💀

**Comment by lludlow** [comment]
well, at least google indexed this so when I searched the error I found some information...

@gitstua any updates? status page still shows all green

**Reply by JellyJoe13** [reply]
Maybe changing the label from question to bug/error would help? Maybe that brings it to the developers attention...
Never opened a discussion here, is that possible @elena-alcaraz ?

**Comment by argmaxin** [comment]
any idea on what's causing this ?

**Reply by JellyJoe13** [reply]
My best guess would be a broken check function before the publishing which aborts the process...

**Reply by argmaxin** [reply]
> My best guess would be a broken check function before the publishing which aborts the process...

perhaps so, and if you had to guess, will this be resolved within a week ?

**Reply by JellyJoe13** [reply]
I have absolutely no idea. I sure hope that they could fix this now as I originally wanted to publish the documentation of my master thesis project^^....
Theoretically rolling back to a previous state would fix the issue but... Not sure whether they will do that or if anyone will work on this before monday. In the first place why roll out a new function on a Friday? (I guess they rolled that out today as there seemed to be no issue before today, pretty sure that someone would have noticed)

**Comment by felipelfv** [comment]
I have the same issue. Yesterday evening and this morning I followed the steps described here (https://ucsb-meds.github.io/creating-quarto-websites/) and here (https://quarto.org/docs/publishing/github-pages.html). At the last step, when trying to change from roots to docs and save, I get the "Invalid GitHub Pages source folder" message

**Reply by argmaxin** [reply]
the worst part is that it does not say anything about what is causing it. Like the reason is so vague.

**Comment by andrewjarkwahwong** [comment]
+1 Im trying to use https://github.com/alshedivat/al-folio. My save button is initially greyed out but if I select gh-pages or any option, I keep getting a "Invalid GitHub Pages source folder."

![image](https://github.com/community/community/assets/136361522/856391fb-f302-4791-bdb4-c15623b06f37)

Repo: https://github.com/andrewjarkwahwong/andrewjarkwahwong.github.io

Any help would be appreciated and glad im not the only one.

**Reply by JellyJoe13** [reply]
Well, if this makes you feel better, I don't really think it is something that we do but rather that something on GitHubs side... Selecting something in two drop down menus almost identically to doing something I did half a year ago does not seem like a me-problem. Plus someone stated that it also does not work for simplest index.html files.
I mean sure you could technically try to put your html files in another folder/branch, however that will most likely not do anything. Or you could use their GitHub Action thingy - which was probably something that was updated and causes the issue now.

The last one actually has a chance of working as noone here complained about that now working yet or it is just not popular^^

**Reply by zhoujiayi003** [reply]
change default branch from master branch to gh-pages branch. Then select gh-pages again

**Comment by JustinRibs** [comment]
same issue

**Comment by jeffreyhsu321** [comment]
If it helps anyone, I ended up deploying docusaurus to gh-pages using the same account as the owner of the repository. I was having the same errors as y'all when I was using a collaborator account to deploy.

**Reply by lludlow** [reply]
I am the owner and still having this issue

**Comment by FroZe36** [comment]
same problem

**Comment by JellyJoe13** [comment]
Ok, I found out a possible workaround with workflows/Actions, those still work.
Prerequisites: Have elements to push to pages on main branch (may be possible on other branches but I did not get it to work) and assuming you have a folder of HTML files in your repo

Steps:
- Settings -> Pages
- GitHub Actions in the source dropdown menu
- Choose template "Static HTML" (configure)
- (It opens a config file to add to repo)
- If your HTML content is in a folder, then change in line 40 "." to the path, e.g. "./docs"
- You might also want to change the trigger condition, e.g. by deleting "push: branches: ["main"], or else every time you push something to main it is executed.
- Commit it, in case you left the trigger condition it should deploy
- Else: Go to Actions -> Deploy static content to pages and hit Run workflow with main branch.

Btw. I would appreciate if someone got this to work so that this is also possible on a non-default branch, I am not a fan of merging my docs and main folder in my repo where I wanted to use pages originally.

Example/Proof: https://github.com/JellyJoe13/PagesTest2 (the 'docs' part in the config does not work, so ignore that one)

**Reply by grtj1417** [reply]
@JellyJoe13 sorry this method is not working for me, I think the only way is to wait until github resolve this problem

**Reply by IhsanKhaliq** [reply]
Worked for me, thank you!

**Comment by sebastian-raubach** [comment]
Same issue here.

**Comment by Ocece77** [comment]
same :c 
<img width="1429" alt="Capture d’écran 2024-06-28 à 18 47 39" src="https://github.com/community/community/assets/132077432/ca626ba7-a938-4e14-aabe-f4625ac52bb1">

**Reply by tangbojin** [reply]
Same issue here.Guys fix it please.

**Reply by tangbojin** [reply]
I chang my branch to the default branch [main].It works.

**Comment by philippemiron** [comment]
I created a GitHub ticket, and from their answer, it looks like it's not going to be fixed. The deploy from a branch option is going away in 2 days, as shown at the top of [creating-a-github-pages-site](https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site).

I'm not sure if there is a workaround because the deploy Github Action is not triggered automatically if you have another Github Action that:
- build website
- copy static files to the gh-pages branch
- create a commit

If you run it manually, it does update correctly. It seems to be related to:

`Commits pushed by a GitHub Actions workflow that uses the GITHUB_TOKEN do not trigger a GitHub Pages build.`

on [publishing-source-for-your-github-pages](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site).

If someone has a solution, that would be helpful 😃

**Reply by VivekMChawla** [reply]
@philippemiron - Thank you for sharing this.

**Reply by tcbyrd** [reply]
>  The deploy from a branch option is going away in 2 days

I'm on the Pages team. There seems to be a bit of confusion on this deprecation notice (which I understand, as it's confusing to even talk about the difference). The deploy from a branch option will remain, it will just require Actions itself to be enabled (at the repository level, not the Pages setting). We're separately looking into why docs sources are having issues. The intent is for that feature to still function as it always has, just that it will build on Actions transparently instead of on our legacy build infrastructure that pre-dates Actions.

**Reply by ManuelBerrueta** [reply]
@tcbyrd are you saying then that the way the GitHub pages were being deployed with the ability to switching the branch is now deprecated, and now it requires us to create our own GitHub action to build/deploy?

If so ,that would be a great answer to clarify the issue and close this thread.

Is there somewhere where it is mentioned that this is happening, it apparently is a surprise to many of us.

**Reply by tcbyrd** [reply]
> @tcbyrd are you saying then that the way the GitHub pages were being deployed with the ability to switching the branch is now deprecated, and now it requires us to create our own GitHub action to build/deploy?

No I'm saying the opposite. The deploy from a branch option is *NOT* going away.

**Reply by ManuelBerrueta** [reply]
So, then the deployment of GitHub Pages should be working as expected and thus there is an issue?

Maybe I misunderstood from what you replied there that it was only going to work for docs now.

**Reply by tcbyrd** [reply]
Yeh we found the issue. Working on getting a fix out now.

**Reply by ManuelBerrueta** [reply]
Got it thanks!

**Comment by kscript** [comment]
![image](https://github.com/community/community/assets/6665066/b3a3d28b-fbd3-4590-924a-b973e8b7eaf4)

**Reply by Kuuuube** [reply]
Thank you! This worked.

**Comment by sinaatalay** [comment]
Hi everyone,

I was having the same issue, but it seems like the way GitHub pages are deployed is changing.

My solution was to give up the `Deploy from a branch` option and use the `GitHub Actions` instead.

I have updated my CI deployment workflow from this:

https://github.com/sinaatalay/rendercv/blob/fafa4084cf5ab4153a1830a9fff1fb1ca791e949/.github/workflows/deploy-docs.yaml

to this:

https://github.com/sinaatalay/rendercv/blob/main/.github/workflows/deploy-docs.yaml

Basically, the new workflow builds the website and then pushes the final HTML files as an artifact. Then, `actions/deploy-pages@v4` takes that artifact and deploys the website.

**Reply by tcbyrd** [reply]
> My solution was to give up the Deploy from a branch option and use the GitHub Actions instead.

Cross referencing my comment in [another thread](https://github.com/orgs/community/discussions/130206#discussioncomment-9908587). To summarize, the deploy from a branch option is _not_ being deprecated. There is a bug we're working on a fix for, but you should not have to move away from using a branch as the source if you don't want to.

**Comment by zhoujiayi003** [comment]
Hi everyone

a solution may solve the problem: change the default branch from main branch to gh-pages branch. 

This solves the problem when I try to use  https://github.com/alshedivat/al-folio

**Reply by CrazyKamel** [reply]
Hi, thank you. I changed my default branch from main to gh-pages, set gh-pages branch for github pages deployment and then, set back main as the default branch.

**Reply by tian1327** [reply]
Thank you so much for sharing this workaround! It works for me!

**Reply by new-AF** [reply]
thx for the workaround

**Reply by VivekMChawla** [reply]
@zhoujiayi003 - You are a lifesaver! Thank you for figuring out this workaround and sharing it with us!

**Comment by new-AF** [comment]
same

**Comment by lavumi** [comment]
It seems that changing the branch and folder isn't working. I am currently using GitHub Pages with the master branch pointing to the docs folder. I tried to switch it to the gh-pages branch pointing to the root directory, but it didn't work.

**Comment by YiMysty** [comment, marked as answer]
👋 A fix has been applied, and the issue should have been resolved. We apologize for any inconvenience caused.

**Reply by ManuelBerrueta** [reply]
Awesome 👏, thank you for the hard work!
I know that's not how anyone wants to end a Friday but I know all the folks really appreciate it!

**Comment by elena-alcaraz** [comment]
It's looks to be fixed :) Thanks to all of you!!!

**Comment by iCodeWebApps** [comment]
I converted over to AWS Static Website hosting as a result of the multi-day outage resulting from this bug.
